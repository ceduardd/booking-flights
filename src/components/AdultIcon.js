const AdultIcon = () => {
  return <i className="fas fa-male"></i>;
};

export default AdultIcon;
