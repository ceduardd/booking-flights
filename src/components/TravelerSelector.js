const TravelerSelector = ({ label, value, changeHandler, icon: Icon, min }) => {
  return (
    <label className="traveler-selector">
      <span>{label}:</span>
      <div style={{ marginTop: '5px' }} className="form-group">
        <input
          className="form-control"
          type="number"
          value={value}
          onChange={e => changeHandler(e.target.value)}
          min={min}
        />
        <Icon />
      </div>
    </label>
  );
};

export default TravelerSelector;
