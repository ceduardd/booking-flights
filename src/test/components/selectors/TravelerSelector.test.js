import '@testing-library/jest-dom';
import '@testing-library/react';
import { shallow } from 'enzyme';
import AdultIcon from '../../../components/icons/AdultIcon';
import TravelerSelector from '../../../components/selectors/TravelerSelector';

describe('<TravelerSelector />', () => {
  test('should be match with snapshot', () => {
    const wrapper = shallow(<TravelerSelector icon={AdultIcon} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should have the correct props', () => {
    const label = 'Adults';

    const changeHandler = jest.fn();

    const wrapper = shallow(
      <TravelerSelector
        label={label}
        icon={AdultIcon}
        changeHandler={changeHandler}
        value={0}
      />
    );

    const input = wrapper.find('.form-control');

    const e = {
      target: {
        value: 1,
      },
    };

    input.simulate('change', e);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('span').text()).toBe(`${label}:`);
    expect(changeHandler).toHaveBeenCalled();
  });
});
