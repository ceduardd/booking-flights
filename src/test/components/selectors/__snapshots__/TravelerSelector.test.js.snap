// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TravelerSelector /> should be match with snapshot 1`] = `
<label
  className="traveler-selector"
>
  <span>
    :
  </span>
  <div
    className="form-group"
    style={
      Object {
        "marginTop": "5px",
      }
    }
  >
    <input
      className="form-control"
      onChange={[Function]}
      type="number"
    />
    <AdultIcon />
  </div>
</label>
`;

exports[`<TravelerSelector /> should have the correct props 1`] = `
<label
  className="traveler-selector"
>
  <span>
    Adults
    :
  </span>
  <div
    className="form-group"
    style={
      Object {
        "marginTop": "5px",
      }
    }
  >
    <input
      className="form-control"
      onChange={[Function]}
      type="number"
      value={0}
    />
    <AdultIcon />
  </div>
</label>
`;
